moveit_core/robot_state/include/moveit/robot_state/robot_state.h

```diff
@@ -1044,122 +1071,133 @@ as the new values that correspond to the group */
+     \e distance. The resulting joint values are stored in the vector \e traj, one by one.
+     The maximum distance in Cartesian space between consecutive points on the resulting path is specified by
+     \e max_step.
+     If a \e validCallback is specified, this is passed to the internal call to setFromIK(). In case of IK failure,
+     the computation of the path stops and the value returned corresponds to the distance that was computed and for
+     which corresponding states were added to the path.  At the end of the function call, the state of the group
+     corresponds to the last attempted Cartesian pose.
+     During the computation of the trajectory, it is usually preferred if consecutive joint values do not 'jump' by a
+     large amount in joint space, even if the Cartesian distance between the corresponding points is small as expected.
+     To account for this, the \e jump_threshold struct is provided, which comprises three fields:
+     \e jump_threshold_factor, \e prismatic_jump_threshold and \e revolute_jump_threshold.
+     If \e prismatic_jump_threshold or \e revolute_jump_threshold are non-zero, we test for absolute jumps.
+     If \e jump_threshold_factor is non-zero, we test for relative jumps. Otherwise (all params are zero), jump
+     detection is disabled.
+     For relative jump detection, the average joint-space distance between consecutive points in the trajectory is
+     computed. If any individual joint-space motion delta is larger then this average distance by a factor of
+     \e jump_threshold_factor, this step is considered a failure and the returned path is truncated up to just
+     before the jump. */
   double computeCartesianPath(const JointModelGroup* group, std::vector<RobotStatePtr>& traj, const LinkModel* link,
                               const Eigen::Vector3d& direction, bool global_reference_frame, double distance,
-                              double max_step, double jump_threshold,
+                              double max_step, const JumpThreshold& jump_threshold,
                               const GroupStateValidityCallbackFn& validCallback = GroupStateValidityCallbackFn(),
                               const kinematics::KinematicsQueryOptions& options = kinematics::KinematicsQueryOptions());
 
+  double computeCartesianPath(const JointModelGroup* group, std::vector<RobotStatePtr>& traj, const LinkModel* link,
+                              const Eigen::Vector3d& direction, bool global_reference_frame, double distance,
+                              double max_step, double jump_threshold_factor,
+                              const GroupStateValidityCallbackFn& validCallback = GroupStateValidityCallbackFn(),
+                              const kinematics::KinematicsQueryOptions& options = kinematics::KinematicsQueryOptions())
+  {
+    return computeCartesianPath(group, traj, link, direction, global_reference_frame, distance, max_step,
```

Comment:
I moved the implementation of these functions into the header, as this clearly states that they simply relay to the actual worker function. Having nice constructors for the the `JumpThreshold` struct facilitates readability.

Reply:
This is great